ui/app/components/secret-engine/configure-aws.ts

```diff
@@ -43,11 +44,34 @@ interface Args {
   leaseConfig: LeaseConfigModel;
   rootConfig: RootConfigModel;
   backendPath: string;
+  issuer?: string;
+}
+
+class IssuerConfig {
+  original = '';
+  @tracked issuer = '';
+  @tracked noRead = false;
+  init(initialIssuer: string | undefined) {
+    if (initialIssuer === 'no-read') {
+      // if we couldn't read it, this string comes back from the route model
+      this.noRead = true;
+    } else if (initialIssuer) {
+      this.original = initialIssuer;
+      this.issuer = initialIssuer;
+    }
+  }
+  // this is to make it work with FormField
+  @action set(_: string, val: string) {
```

Comment:
Do you need `@action` here?

Reply:
I believe so, but I can play around with it

Reply:
commenting so I can learn here as well (or correct my understanding), but if I remember right the action decorator binds the function directly to the instance of the class it's called from. So by calling `this` in the action, you know that `this` refers to the class it's within. I believe this is a bigger deal when you're passing actions around, but please correct me if that's wrong.

Reply:
Good memory Angel, and well said! I believe this is the case

Reply:
Awesome! But this is a javascript class, right? I thought the `@action` decorator was specific to just ember components 🤔

Reply:
Pretty sure components have evolved to be very thin wrappers around classes, and decorators are pretty class-agnostic. I can't find specific docs on this, but from [here](https://guides.emberjs.com/release/upgrading/current-edition/native-classes/#toc_decorators) we see reference to the binding that Angel mentioned:

> The action decorator also binds actions, so you can refer to them directly in templates without the {{action}} helper:

and [here](https://www.pzuraq.com/blog/do-you-need-ember-object#myth-3-event-listeners-only-work-with-classes-that-extend-from-emberobject) we see that an action is applied on a regular old class. Elsewhere in this post we see another reference to the point of `@action`, which is binding `this` correctly to the class:

> Note that we’re using the @action decorator here to bind the context of the someObserver method, so it’s this always refers to the class instance.

Reply:
This method specifically is being passed to `<FormField />` and called from the class so the decorator is extraneous, but if and when we update this to use HDS components directly or otherwise calling this update from the template, we will need the `@action` decorator